### PR TITLE
[SpecDecode] Fix overlap-path correctness for idle, prefill gating, and non-greedy verify

### PR DESCRIPTION
**Pull request by lingebeng**
## Summary
  This PR addresses the five review items around speculative overlap correctness and dispatch safety:

  1. Remove the EAGLE-local idle fallback so idle batches stay on the scheduler/model-runner idle path.
  2. Add an explicit assert for `custom_mask.dtype != jnp.bool_`.
  3. Document the current grammar / vocab_mask / penalty gap in the spec decode overlap gate.
  4. Prevent non-fusable spec prefill batches from entering the fused overlap entry and crashing the server.
  5. Pass per-request sampling params through fused overlap verify so non-greedy batches no longer reuse request 0's params.

## Test results

**MiMo-V2-Flash: greedy vs non-greedy (jax 0.8.1 / random 16384→4096)**

BS | Mode | Output | Input | Total | Peak | TTFT (ms) | ITL (ms) | P99 ITL (ms)
-- | -- | -- | -- | -- | -- | -- | -- | --
32 | greedy | 1704.7 | 6816.9 | 8521.6 | 4747 | 15349 | 14.62 | 40.12
32 | non-greedy | 1389.9 | 5557.9 | 6947.8 | 4445 | 13352 | 18.85 | 41.72
64 | greedy | 1821.6 | 7283.9 | 9105.5 | 5889 | 31292 | 26.72 | 58.23
64 | non-greedy | 1684.3 | 6735.0 | 8419.3 | 5817 | 35250 | 28.47 | 81.92
128 | greedy | 1993.5 | 7971.6 | 9965.1 | 7454 | 67620 | 46.84 | 483.50
128 | non-greedy | 1861.7 | 7444.4 | 9306.1 | 6798 | 67952 | 51.13 | 488.36

### AIME24 accuracy
| Mode       | Sampling             | Score      | Correct |
| ---------- | -------------------- | ---------- | ------- |
| greedy     | temp=0.0             | **0.8333** | 25/30   |
| non-greedy | temp=1.0, top_p=0.95 | **0.8333** | 25/30   |

### CODE REVIEW

File: python/sgl_jax/srt/managers/scheduler.py [hunk 1]
```diff
@@ -2069,6 +2069,17 @@ def _run_speculative_batch(
...
+            # Authoritative gate: the cheap can_use_* pre-filter above cannot see
+            # the merged sampling_info or the worker's NEXTN/topk/num_steps config,
+            # so confirm against the same predicate the fused prefill entry uses.
+            # Otherwise forward_batch_speculative_prefill_overlap would raise
+            # NotImplementedError (penalty/vocab_mask, or non-fusable spec config)
+            # and crash the server; falling back here routes the batch to the
+            # normal forward_batch_speculative_generation path instead.
+            use_spec_prefill_overlap = self.draft_worker._can_use_fused_spec_prefill(
+                model_worker_batch
+            )
```
Comment (pengchengneo) [inline python/sgl_jax/srt/managers/scheduler.py]: use_spec_prefill_overlap = can_use_spec_prefill_overlap(
            self.enable_overlap, self.spec_algorithm, batch
        ) and self.draft_worker._can_use_fused_spec_prefill(
                model_worker_batch
            )
